[댓글 기능] 르윈 미션 제출합니다

yuyu154 opened the pull request:
1. 페어의 코드와 merge해서 제출했습니다
2. sample data를 "data.sql" 안에 세팅했습니다

yk1028 committed:
feat: 실습

yk1028 committed:
feat: 환경설정

yk1028 committed:
feat: HTML 중복제거

yk1028 committed:
refactor: thymeleaf-layout 제거

yk1028 committed:
feat: static resource path 수정

yk1028 committed:
fix: devtools 개발 시에만 사용하도록 변경

yk1028 committed:
feat: 게시물 생성

yk1028 committed:
feat: 게시글 생성 후 전체 리스트 조회

yk1028 committed:
feat: 게시글 개별 조회

yk1028 committed:
style: formatting

yk1028 committed:
feat: Article에 id 추가

yk1028 committed:
feat: 게시글 수정 페이지로 이동

yk1028 committed:
feat: 게시글 수정

yk1028 committed:
feat: 게시글 삭제

yk1028 committed:
test: 테스트간 영향이 없도록 수정

yk1028 committed:
fix: UI 버그 수정

yk1028 committed:
style: formatting

yk1028 committed:
test: 테스트 코드 중복 제거

yk1028 committed:
[게시글 생성/조회/수정/삭제] 히브리 미션 제출합니다. (#16)

* feat: 실습

* feat: 환경설정

* feat: HTML 중복제거

* refactor: thymeleaf-layout 제거

* feat: static resource path 수정

* fix: devtools 개발 시에만 사용하도록 변경

* feat: 게시물 생성

* feat: 게시글 생성 후 전체 리스트 조회

* feat: 게시글 개별 조회

* style: formatting

* feat: Article에 id 추가

* feat: 게시글 수정 페이지로 이동

* feat: 게시글 수정

* feat: 게시글 삭제

* test: 테스트간 영향이 없도록 수정

* fix: UI 버그 수정

* style: formatting

* test: 테스트 코드 중복 제거

yk1028 committed:
test: ArticleRepositoryTests에서 서버를 띄우지 않도록 수정

yk1028 committed:
refactor: ArticleRepository 클래스의 clear 메서드 제거

yk1028 committed:
Merge remote-tracking branch 'woowacourse/yk1028' into yk1028

yk1028 committed:
refactor: Article id의 동시성 보장

yk1028 committed:
refactor: matchId 구현

yk1028 committed:
refactor: next id를 Article에서 관리하도록 수정

yk1028 committed:
test: Article test 추가

yk1028 committed:
refactor: @Autowired를 생성자 주입으로 변경

yk1028 committed:
refactor: JAP 활용 방식으로 변경

yk1028 committed:
feat: sql formatting 설정

yk1028 committed:
test: test 코드 수정

yk1028 committed:
docs: 기능목록 작성

yk1028 committed:
refactor: matchId 메서드 삭제

yk1028 committed:
feat: DB를 H2에서 MySQL로 변경

yk1028 committed:
feat: Index 페이지에 대한 처리를 IndexController에서 담당하도록 수정

yk1028 committed:
feat: Login 페이지로 이동하는 기능 구현

UserController에서 유저와 관련된 control을 담당하도록 하였습니다.

yk1028 committed:
feat: 회원가입 페이지로 이동하는 기능 구현

yk1028 committed:
feat: 회원가입 기능 구현

yk1028 committed:
feat: email 중복 검사

yk1028 committed:
feat: email 중복 시 errorMessage를 Model에 담아서 페이지에 노출

yk1028 committed:
feat: sign-up 페이지에서 좌측 상단 로고 누르면 홈으로 이동하도록 수정

yk1028 committed:
feat: 이름 규칙 적용

이름은 2~10자로 제한하며 숫자나 특수문자가 포함될 수 없다.

yk1028 committed:
feat: 비밀번호 규칙 적용

비밀번호는 8자 이상의 소문자, 대문자, 숫자, 특수문자의 조합이다.

yk1028 committed:
feat: 비밀번호 확인 기능 구현

yk1028 committed:
feat: 유저 목록 페이지 구현

yk1028 committed:
feat: FE에서 비밀번호 검증 구현

yk1028 committed:
feat: Login 기능을 User로부터 분리

yk1028 committed:
feat: http header/body logging

yk1028 committed:
feat: Login 기능 구현

yk1028 committed:
feat: 로그인 실패 시 상황에 맞는 실패 메시지를 띄운다.

- 이메일이 없는 경우
- 비밀번호가 틀린 경우

yk1028 committed:
feat: 로그아웃 기능 구현

yk1028 committed:
feat: 로그인 한 유저가 로그인/회원가입 화면에 접근할 경우 메인 화면을 띄운다.

yk1028 committed:
feat: 마이 페이지 구현

yk1028 committed:
feat: mypage-edit으로 이동하는 기능 구현

yk1028 committed:
feat: User 이름 수정 기능 구현

yk1028 committed:
feat: LogOut 상태에서 article을 생성할 수 없도록 수정

yk1028 committed:
feat: login 상태에서만 article 작성 가능하도록 수정

yk1028 committed:
feat: 자신의 Article만 수정 가능하도록 구현

yk1028 committed:
feat: 자신의 Article만 삭제 가능하도록 구현

yk1028 committed:
feat: id에 해당하는 User의 mypage를 보여주도록 수정

yk1028 committed:
feat: 자신의 회원 정보만 수정할 수 있도록 구현

yk1028 committed:
feat: 회원 탈퇴기능 구현

yk1028 committed:
refactor: Optional.get() 사용 부분 변경

yk1028 committed:
feat: User 이름 수정시 유효성 검사

yk1028 committed:
refactor: User 이름 수정시 유효성 검사 예외처리 방법 수정

UserArgumentException class 생성

yk1028 committed:
refactor: UserRepository에서 조회 결과를 Optional로 반환하도록 수정

yk1028 committed:
test: User test 추가

- edit, delete에 대한 테스트를 추가하였습니다.
- h2 db로 변경 및 초기 데이터 추가

yk1028 committed:
style: formatting

yk1028 committed:
fix: gradle.build 오타 수정

yk1028 committed:
refactor: User의 이름, 비밀번호에 대한 유효성 검사를 service에서 domain으로 변경

yk1028 committed:
feat: User 삭제시 예외처리 추가

yk1028 committed:
refactor: Pattern을 static으로 변경

yk1028 committed:
test: LogInServiceTest 수정

yk1028 committed:
refactor: default 생성자를 private로 변경

yk1028 committed:
refactor: mysql-connector-java runtimeOnly로 변경

yk1028 committed:
refactor: ArticleService 구현

yk1028 committed:
test: LogInServiceTest 수정

yk1028 committed:
test: Controller에서 사용되는 Repository를 Service로 변경

yk1028 committed:
refactor: UserArticleMissmatchException 위치 변경

yk1028 committed:
refactor: ControllerAdvice 적용

yk1028 committed:
refactor: ControllerAdvice 추가 적용

yk1028 committed:
refactor: LogInService 수정

yk1028 committed:
refactor: updateArticle 메서드 수정

yk1028 committed:
refactor: build.gradle 수정

yk1028 committed:
refactor: ControllerAdvice를 ControllerExceptionAdvice로 이름 변경

yk1028 committed:
refactor: Dto 패키지 위치를 Service 하위로 변경

yk1028 committed:
[게시판 회원 기능] 히브리 미션 제출합니다. (#84)

* feat: 실습

* feat: 환경설정

* feat: HTML 중복제거

* refactor: thymeleaf-layout 제거

* feat: static resource path 수정

* fix: devtools 개발 시에만 사용하도록 변경

* feat: 게시물 생성

* feat: 게시글 생성 후 전체 리스트 조회

* feat: 게시글 개별 조회

* style: formatting

* feat: Article에 id 추가

* feat: 게시글 수정 페이지로 이동

* feat: 게시글 수정

* feat: 게시글 삭제

* test: 테스트간 영향이 없도록 수정

* fix: UI 버그 수정

* style: formatting

* test: 테스트 코드 중복 제거

* test: ArticleRepositoryTests에서 서버를 띄우지 않도록 수정

* refactor: ArticleRepository 클래스의 clear 메서드 제거

* refactor: Article id의 동시성 보장

* refactor: matchId 구현

* refactor: next id를 Article에서 관리하도록 수정

* test: Article test 추가

* refactor: @Autowired를 생성자 주입으로 변경

* refactor: JAP 활용 방식으로 변경

* feat: sql formatting 설정

* test: test 코드 수정

* docs: 기능목록 작성

* refactor: matchId 메서드 삭제

* feat: DB를 H2에서 MySQL로 변경

* feat: Index 페이지에 대한 처리를 IndexController에서 담당하도록 수정

* feat: Login 페이지로 이동하는 기능 구현

UserController에서 유저와 관련된 control을 담당하도록 하였습니다.

* feat: 회원가입 페이지로 이동하는 기능 구현

* feat: 회원가입 기능 구현

* feat: email 중복 검사

* feat: email 중복 시 errorMessage를 Model에 담아서 페이지에 노출

* feat: sign-up 페이지에서 좌측 상단 로고 누르면 홈으로 이동하도록 수정

* feat: 이름 규칙 적용

이름은 2~10자로 제한하며 숫자나 특수문자가 포함될 수 없다.

* feat: 비밀번호 규칙 적용

비밀번호는 8자 이상의 소문자, 대문자, 숫자, 특수문자의 조합이다.

* feat: 비밀번호 확인 기능 구현

* feat: 유저 목록 페이지 구현

* feat: FE에서 비밀번호 검증 구현

* feat: Login 기능을 User로부터 분리

* feat: http header/body logging

* feat: Login 기능 구현

* feat: 로그인 실패 시 상황에 맞는 실패 메시지를 띄운다.

- 이메일이 없는 경우
- 비밀번호가 틀린 경우

* feat: 로그아웃 기능 구현

* feat: 로그인 한 유저가 로그인/회원가입 화면에 접근할 경우 메인 화면을 띄운다.

* feat: 마이 페이지 구현

* feat: mypage-edit으로 이동하는 기능 구현

* feat: User 이름 수정 기능 구현

* feat: LogOut 상태에서 article을 생성할 수 없도록 수정

* feat: login 상태에서만 article 작성 가능하도록 수정

* feat: 자신의 Article만 수정 가능하도록 구현

* feat: 자신의 Article만 삭제 가능하도록 구현

* feat: id에 해당하는 User의 mypage를 보여주도록 수정

* feat: 자신의 회원 정보만 수정할 수 있도록 구현

* feat: 회원 탈퇴기능 구현

* refactor: Optional.get() 사용 부분 변경

* feat: User 이름 수정시 유효성 검사

* refactor: User 이름 수정시 유효성 검사 예외처리 방법 수정

UserArgumentException class 생성

* refactor: UserRepository에서 조회 결과를 Optional로 반환하도록 수정

* test: User test 추가

- edit, delete에 대한 테스트를 추가하였습니다.
- h2 db로 변경 및 초기 데이터 추가

* style: formatting

* fix: gradle.build 오타 수정

* refactor: User의 이름, 비밀번호에 대한 유효성 검사를 service에서 domain으로 변경

* feat: User 삭제시 예외처리 추가

* refactor: Pattern을 static으로 변경

* test: LogInServiceTest 수정

* refactor: default 생성자를 private로 변경

* refactor: mysql-connector-java runtimeOnly로 변경

* refactor: ArticleService 구현

* test: LogInServiceTest 수정

* test: Controller에서 사용되는 Repository를 Service로 변경

* refactor: UserArticleMissmatchException 위치 변경

* refactor: ControllerAdvice 적용

* refactor: ControllerAdvice 추가 적용

* refactor: LogInService 수정

* refactor: updateArticle 메서드 수정

* refactor: build.gradle 수정

* refactor: ControllerAdvice를 ControllerExceptionAdvice로 이름 변경

* refactor: Dto 패키지 위치를 Service 하위로 변경

yk1028 committed:
Merge remote-tracking branch 'woowacourse/yk1028' into yk1028

yk1028 committed:
docs: 기능 목록 추가

yk1028 committed:
refactor: user, article 패키지 분리

- 패키지 안에 관련 domain, repository 포함

yk1028 committed:
refactor: 로그인 할 때 email과 password의 검증 과정 분리

yk1028 committed:
refactor: UserDeleteException.class 메시지 수정

yk1028 committed:
refactor: User.class 메소드명 수정

- setName() -> updateName()

yk1028 committed:
refactor: 유저 일치 여부에 따른 게시글 처리 방법 수정

- 수정 페이지 접근, 게시글 수정, 게시글 삭제

yk1028 committed:
refactor: User.class <-> Article.class 양방향 맵핑

- Article.class의 userId(Long) -> author(User)로 수정
- User.class에 articles(List<Article>) 추가

yk1028 committed:
refactor: ArticleService.class 의 save 메소드 수정

yk1028 committed:
feat: comment 생성 구현

- service

yk1028 committed:
feat: Article에 등록된 모든 Comment 조회 구현

yk1028 committed:
feat: comment controller 및 view 구현

- 생성, 조회

yk1028 committed:
feat: comment update 기능 구현

- service, controller

yk1028 committed:
feat: comment delete 기능 구현

- service, controller

yk1028 committed:
feat: comment delete 기능 구현

- view

yk1028 committed:
feat: comment update View 구현

yk1028 committed:
refactor: CommentService.class deleteAll 메서드 삭제

yk1028 committed:
refactor: domain에 validate annotation 추가

yk1028 committed:
test: test 추가

- Article, User 삭제시 연관된 Comment도 삭제되는지 테스트

yuyu154 committed:
Revert "[게시판 회원 기능] 르윈 미션 제출합니다 (#59)"

This reverts commit f5d67730597d6bec8dec209d8b6cfc28c8f4aac4.

yuyu154 committed:
merge: 페어 프로그래밍 결과물 merge

yuyu154 committed:
refacotr. IndexControllerTest.class 삭제

yuyu154 committed:
hotfix: Merge 이후 중복된 부분 삭제

yuyu154 committed:
test : IndexControllerTest.class 추가

yuyu154 committed:
refactor : Test.class 메소드 public으로 수정

kang-hyungu reviewed (changes requested):
안녕하세요 르윈님 리뷰어 강현구입니다
코드 구현하느라 고생하셨어요
커밋 갯수가 엄청나네요 :open_mouth: 
몇 가지 피드백 남겼으니 확인해보시고 적용해보세요!

yuyu154 committed:
refactor : 피드백 반영

1. @Column length 속성 추가
2. FetchType LAZY로 수정

yuyu154 committed:
refactor : 피드백 반영 2

1. getter -> 메시지 보내는 방식으로 수정 (ArticleService.class)
2. HttpServletRequest -> HttpSession
3. getter -> 메시지 보내는 방식으로 수정 (ArticleController.class)
4. build.gradle 수정

yuyu154 committed:
refactor : CommentService.class update, delete 메소드 수정

yuyu154 committed:
refactor : CommentService.class save() 수정

yuyu154 committed:
refactor : LoginChecker.class 추가 (로그인 확인, 인가 기능)

yuyu154 committed:
refactor : UserArgumentException.class 처리 로직 추가

yuyu154 committed:
hotfix : 쓰이지 않는 메소드 제거

yuyu154 committed:
hotfix : try-catch 문 구조 수정

yuyu154 committed:
refactor : UserService.class에 findUserPublicInfoByArticle() 추가

yuyu154 commented:
안녕하세요! 
피드백 반영하는 중에 질문이 있어 코멘트 올립니다!

피드백 반영하다가 Getter를 쓰는 부분이 굉장히 많다고 느꼈습니다.
컨트롤러에서 DTO를 서비스 객체로 전달하는 방식을 쓰고 있는데요.  
서비스 객체 내부에서 DTO의 getter (ex. getId, getUserId) 를 쓰면서 비교할 수 밖에 없었습니다.  
DTO에는 setter, getter를 제외한 로직을 넣는 것이 좋지 않다고 알고 있어 메소드를 추가하기도 애매하다고 생각했습니다.   

이런 상황에는  
1. DTO를 도메인 객체로 변환하고 비교 로직을 쓰는게 좋을까요
2. 아니면 지금처럼 DTO를 비교할때는 getter를 쓰는게 좋을까요?

yuyu154 committed:
refactor : ArticleService.class save() 수정

yuyu154 committed:
refactor : LoginController.class 수정

yuyu154 committed:
refactor : ArticleVo.class 삭제

yuyu154 committed:
refactor : UserService.class update() 함수 변경

- 인자 자료형 변경

yuyu154 committed:
refactor : Controller, Service Class 인자 UserPublicInfoDto -> UserSessionDto

kang-hyungu commented:
> 안녕하세요!
> 피드백 반영하는 중에 질문이 있어 코멘트 올립니다!
> 
> 피드백 반영하다가 Getter를 쓰는 부분이 굉장히 많다고 느꼈습니다.
> 컨트롤러에서 DTO를 서비스 객체로 전달하는 방식을 쓰고 있는데요.
> 서비스 객체 내부에서 DTO의 getter (ex. getId, getUserId) 를 쓰면서 비교할 수 밖에 없었습니다.
> DTO에는 setter, getter를 제외한 로직을 넣는 것이 좋지 않다고 알고 있어 메소드를 추가하기도 애매하다고 생각했습니다.
> 
> 이런 상황에는
> 
>     1. DTO를 도메인 객체로 변환하고 비교 로직을 쓰는게 좋을까요
> 
>     2. 아니면 지금처럼 DTO를 비교할때는 getter를 쓰는게 좋을까요?

저는 getter를 덜쓰려고 dto에서 도메인 객체로 변환해서 쓰는 편이에요
dto로 전달해서 쓰는게 더 편한 경우도 있으니 상황에 맞춰 선택하세요!

kang-hyungu reviewed (approved):
피드백 반영하신 부분 확인했습니다!
피드백 깔끔하게 잘 반영하셨네요 👍 
질문 주신 내용도 답변 남겼으니 확인해보세요
코드 구현하느라 수고하셨어요!
